### ISSUE TITLE
Feature Request: optionally return 404 when protected files exist but auth fails

### ISSUE THREAD

**Issue by rubiojr** [pocketbase/pocketbase#5208]
When accessing a protected file with an invalid token, the API returns a 403:

```
❯ curl http://localhost:8090/api/files/11h2ewlken3gnc0/de0t70pp59xyfz7/my-private-file.doc?token=foobar
{"code":403,"message":"Insufficient permissions to access the file resource.","data":{}}
```

[This currently makes clear the resource exists](https://meta.stackexchange.com/questions/258756/what-is-the-reason-behind-marking-forbidden-pages-as-404), making fingerprinting possible.

Presumably changing this to a 404 by default is not an option for compatibility or other reasons, but it'd be nice if returning a 404 could be enabled optionally.

**Comment by ganigeorgiev** [comment]
I've just changed it in the refactored version since I'm currently working on the tests for the new file APIs. It will be available with the next v0.23.0 release (there are no ETAs yet) but even as it is file fingerprinting is very difficult because by default all uploaded files have a random part.

**Comment by rubiojr** [comment]
> I've just changed it in the refactored version since I'm currently working on the tests for the new file APIs

Superb 💟 

>  but even as it is file discovery is very difficult because by default all uploaded filenames have a random part (and you need to know from somewhere the targeted record id in the first place).

👍 agreed.

Taking the chance to send my warmest thanks for such an awesome, simple, pain-free and solid software.
